src/string/grapheme/grapheme.cr

```diff
@@ -0,0 +1,258 @@
+      break?(Property.from(c1), Property.from(c2))
+    end
+
+    # :nodoc:
+    #
+    # Returns whether there is a grapheme break between boundclasses lbc and tbc.
+    #
+    # Please note that evaluation of GB10 (grapheme breaks between emoji zwj sequences)
+    # and GB 12/13 (regional indicator code points) require knowledge of previous characters
+    # which is not handled by this oberload. This may result in an incorrect break before
+    # an E_Modifier class codepoint and an incorrectly missing break between two
+    # REGIONAL_INDICATOR class code points if such support does not exist in the caller.
+    #
+    # The rules are graphically displayed in a table on https://www.unicode.org/Public/13.0.0/ucd/auxiliary/GraphemeBreakTest.html
+    #
+    # The implementation is insipred by https://github.com/JuliaStrings/utf8proc/blob/462093b3924c7491defc67fda4bc7a27baf9b088/utf8proc.c#L261
+    def self.break?(lbc : Property, tbc : Property) : Bool
+      return true if lbc.start?                                                   # GB1
+      return false if lbc.cr? && tbc.lf?                                          # GB3
+      return true if lbc.cr? || lbc.lf? || lbc.control?                           # GB4
+      return true if tbc.cr? || tbc.lf? || tbc.control?                           # GB5
+      return false if lbc.l? && (tbc.l? || tbc.v? || tbc.lv? || tbc.lvt?)         # GB6
+      return false if (lbc.lv? || lbc.v?) && (tbc.v? || tbc.t?)                   # GB7
+      return false if (lbc.lvt? || lbc.t?) && tbc.t?                              # GB8
+      return false if tbc.extend? || tbc.zwj?                                     # GB9
+      return false if tbc.spacing_mark?                                           # GB9a
+      return false if lbc.prepend?                                                # GB9b
+      return false if lbc.extended_plus_zero_width? && tbc.extended_pictographic? # GB11 (requires additional handling)
+      return false if lbc.regional_indicator? && tbc.regional_indicator?          # GB12/13 (requires additional handling)
+      true                                                                        # GB999
+    end
+
+    # :nodoc:
+    def self.break?(c1 : Char, c2 : Char, state : Pointer(Property)) : Bool
+      break?(Property.from(c1), Property.from(c2), state)
+    end
+
+    # :nodoc:
+    #
+    # Returns whether there is a grapheme break between boundclasses lbc and tbc.
+    #
+    # Please note that evaluation of GB10 (grapheme breaks between emoji zwj sequences)
+    # and GB 12/13 (regional indicator code points) require knowledge of previous characters
+    # which is accounted for in the state argument.
+    #
+    # The implementation is inspired by https://github.com/JuliaStrings/utf8proc/blob/462093b3924c7491defc67fda4bc7a27baf9b088/utf8proc.c#L291
+    def self.break?(lbc : Property, tbc : Property, state : Pointer(Property)) : Bool
```

Comment:
Not a big deal, but would it be possible to have a more functional implementation of this function (that is, avoid the `Pointer` here)?

Reply:
I guess so. The return value could be a `Tuple(Bool, Property)`. Not sure if that would be significantly better.